cmd/kubelet/app/server.go

```diff
@@ -206,6 +206,10 @@ HTTP server: The kubelet can also listen for HTTP and respond to a simple API
 				klog.Fatal(err)
 			}
 
+			if (kubeletConfig.KubeletCgroups == "" && kubeletConfig.KubeReservedCgroup != "") || (kubeletConfig.KubeletCgroups != kubeletConfig.KubeReservedCgroup) {
```

Comment:
It should be fine for them not to be equal.  For example, I may want to limit both kubelet and the container runtime with kube-reserved, but keep kubelet in its own cgroup.  E.g:
```
/
/kube-reserved
/kube-reserved/kubelet
/kube-reserved/runtime
```

I would specify `--kube-reserved-cgroup=/kube-reserved` and `--kubelet-cgroups=/kube-reserved/kubelet`.

Maybe we can warn when the kubelet is not placed _within_ the kube-reserved cgroup?

Reply:
@dashpole hi, i've made the modification of my configuration:
![image](https://user-images.githubusercontent.com/20693362/50627001-75dfda00-0f6c-11e9-81cb-7970f6e75625.png)
create a directory "kube.service":
![image](https://user-images.githubusercontent.com/20693362/50627086-ca835500-0f6c-11e9-886a-c56c6e08b7fc.png)
and get this:
![image](https://user-images.githubusercontent.com/20693362/50627322-f18e5680-0f6d-11e9-99ee-0f0e0af0b8a9.png)
that's mean "kubeReserved" have come into effect.then i set " --kubelet-cgroups=/kube.service/kubelet.slice":
![image](https://user-images.githubusercontent.com/20693362/50627459-7c6f5100-0f6e-11e9-9511-c1feec42e308.png)
that's cpu.shares and pid for kubelet:
![image](https://user-images.githubusercontent.com/20693362/50627527-bdfffc00-0f6e-11e9-9aa8-ff393ff74d73.png)
obviously,even if i put kubelet within the  kube-reserved cgroup ,the value of "kubeReserved" cannot affect the kubelet.  i mean i cannot control cpu.shares for kubelet by this way.

maybe i misunderstood it，could you give more suggestion？thanks

Reply:
> obviously,even if i put kubelet within the kube-reserved cgroup ,the value of "kubeReserved" cannot affect the kubelet. i mean i cannot control cpu.shares for kubelet by this way.

See [these cgroup docs](https://kernel.googlesource.com/pub/scm/linux/kernel/git/glommer/memcg/+/cpu_stat/Documentation/cgroups/cpu.txt):

"cpu.shares: The weight of each group living in the same hierarchy, that translates into the amount of CPU it is expected to get. Upon cgroup creation, each group gets assigned a default of 1024. The percentage of CPU assigned to the cgroup is the value of shares divided by the sum of all shares in all cgroups in the same level."

So setting shares in `/kube.service` means _everthing_ in that subtree gets that portion of shares relative to other cgroups in `/`.  The `/kube.service/kubelet.slice/cpu.shares` file could allow you to _further_ subdivide cpu shares between cgoups in `/kube.service`.

Reply:
@dashpole ,that's best answer. now i totally understand，thanks for your patience. that's very helpful. this weekend maybe i'll make the modification as your suggestions.

Reply:
i've added warning when the kubelet is not placed within the kube-reserved cgroup